opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/NonDominatedFronts.java

```diff
@@ -0,0 +1,192 @@
+package org.opt4j.optimizers.ea;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.opt4j.core.Individual;
+import org.opt4j.core.Objective;
+import org.opt4j.core.Objectives;
+
+/**
+ * The {@link NonDominatedFronts} sorts each evaluated {@link Individual} into
+ * fronts based on the number of other individuals it is dominated by. The first
+ * front consists of points that are not dominated at all and so on.
+ * 
+ * @author Fedor Smirnov
+ *
+ */
+public class NonDominatedFronts extends ArrayList<List<Individual>> {
```

Comment:
I do not think, this class should expose that it technically uses an arraylist<list<indivudial>> to the outside. Aren't there a lot of arraylist functions (or even all of them?), which make no sense on an object called "NonDominatedFronts"? Isn't this class more an algorithm instead of a data structure?

As I do not have good proposal how to restructure this one, this is not a blocker for merging. Just something I would ask you to rethink.

Reply:
I agree. The user should not be able to manipulate the list. I changed the class accordingly